src/SelfUpdateCommand.php

```diff
@@ -32,7 +32,8 @@ class SelfUpdateCommand extends Command
     public function __construct($applicationName = null, $currentVersion = null, $gitHubRepository = null)
     {
         $this->applicationName = $applicationName;
-        $this->currentVersion = $currentVersion;
+        $version_parser = new VersionParser();
+        $this->currentVersion = $version_parser->normalize($currentVersion);
```

Comment:
I'm not sure if this is necessary or not. Based on my reading of the code, without normalization here, passing a three-party version (2.8.6) would cause version comparisons to fail later on. But the documentation says three-part versions are accepted 🤷

Reply:
Yeah on further testing, this is necessary. If you pass a three-part version to the constructor it may not be compared correctly by the version parser. Basically, everything needs to be normalized or nothing should be.